great_expectations/validator/validator.py

```diff
@@ -515,96 +612,89 @@ def graph_validate(
                         exception_info=exception_info,
                         expectation_config=configuration,
                     )
                     evrs.append(result)
-                return evrs
-            else:
-                raise err
 
-        for configuration in processed_configurations:
-            try:
-                result = configuration.metrics_validate(
-                    metrics,
-                    execution_engine=self._execution_engine,
-                    runtime_configuration=runtime_configuration,
-                )
-                evrs.append(result)
-            except Exception as err:
-                if catch_exceptions:
-                    exception_traceback = traceback.format_exc()
-                    exception_message = str(err)
-                    exception_info = ExceptionInfo(
-                        **{
-                            "exception_traceback": exception_traceback,
-                            "exception_message": exception_message,
-                        }
-                    )
-                    result = ExpectationValidationResult(
-                        success=False,
-                        exception_info=exception_info,
-                        expectation_config=configuration,
-                    )
-                    evrs.append(result)
-                else:
-                    raise err
+                if configuration not in rejected_configurations:
+                    rejected_configurations.append(configuration)
+
+        # Exclude all rejected expectations from list of expectations cleared for validation.
+        for configuration in rejected_configurations:
+            processed_configurations.remove(configuration)
+
+        return evrs, processed_configurations
+
+    def _catch_exceptions_in_failing_expectation_validations(
+        self,
+        exception_traceback: traceback.TracebackException,
+        exception: Exception,
+        failing_expectation_configurations: List[ExpectationConfiguration],
+        evrs: List[ExpectationValidationResult],
+    ) -> List[ExpectationValidationResult]:
+        """
+        Catch exceptions in failing Expectation validations and convert to unsuccessful ExpectationValidationResult
+        Args:
+            exception_traceback: Traceback related to raised Exception
+            exception: Exception raised
+            failing_expectation_configurations: ExpectationConfigurations that failed
+            evrs: List of ExpectationValidationResult objects to append failures to
+
+        Returns:
+            List of ExpectationValidationResult objects with unsuccessful ExpectationValidationResult objects appended
+        """
+        exception_message = str(exception)
+        exception_info = ExceptionInfo(
+            **{
+                "exception_traceback": exception_traceback,
+                "exception_message": exception_message,
+            }
+        )
```

Comment:
More for my edification than a review comment but why `**kwargs` instead of explicit keyword or positional arguments?

I've seen this approach a few times in our codebase so I'm trying to get a better understand of where to apply it. If we only have two arguments here, do we need this pattern? Does it matter either way? Would love to know your thoughts and understand a general rule of thumb here.

Reply:
This is just a copy paste, but you are correct - this should reference the explicit arguments. Will change.